pulp_ansible/app/viewsets.py

```diff
@@ -290,6 +320,45 @@ def sync(self, request, pk):
         )
         return OperationPostponedResponse(result, request)
 
+    @extend_schema(
+        description="Trigger an asynchronous task to sign Ansible content.",
+        responses={202: AsyncOperationResponseSerializer},
+    )
+    @action(detail=True, methods=["post"], serializer_class=AnsibleRepositorySignatureSerializer)
+    def sign(self, request, pk):
+        """
+        Dispatches a sync task.
+        """
+        content_units = {}
+
+        repository = self.get_object()
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        signing_service = serializer.validated_data["signing_service"]
+        content = serializer.validated_data["content_units"]
+
+        if "*" in content:
+            content_units = ["*"]
```

Comment:
This is unfortunate. `content_units` used to be a `dict`.

Reply:
I don't know what you mean. I thought it was always a `list` of strings and if at any point there is an asterisk in it the list would switched to only be an asterisk because mixed behavior wouldn't make sense.

Reply:
Oh, now i see. The confusion is contained to this function. I read the line 332 `content_units = {}` and thought the argument to the task was a `dict`. But you are converting it back to a list in 349 `content_units = list(content_units.keys())`. Why are we not passing the already extracted `content_unit_pks`?

Reply:
To be fair, i see the same pattern of redefining the from dict to list in the modify action in pulpcore.

Reply:
Yes this was confusing, I changed the variable names so `content_units` is always a `dict` and `content_units_pks` is always a `list`.